server/src/main/java/io/crate/execution/dsl/projection/builder/InputColumns.java

```diff
@@ -262,6 +263,19 @@ public Symbol visitReference(Reference ref, SourceSymbols sourceSymbols) {
         return inputColumn;
     }
 
+    @Override
+    public Symbol visitVoidReference(VoidReference ref, SourceSymbols sourceSymbols) {
+        var symbol = visitReference(ref, sourceSymbols);
+        if (symbol.equals(ref)) {
+            for (var e : sourceSymbols.inputs.entrySet()) {
+                var sourceSymbol = e.getKey();
+                if (sourceSymbol.toColumn().equals(ref.toColumn()) && sourceSymbol.valueType().equals(ref.valueType())) {
+                    return e.getValue();
+                }
+            }
+        }
+        return symbol;
+    }
```

Comment:
`select obj_dynamic['u'] from (select obj_dynamic['u'] from t) t2;`
The sourceSymbols.inputs for above query is `obj_dynamic['u']`(ScopedSymbol) mapped to INPUT(0) which cannot match the ref which is a VoidReference.